sql/src/main/java/org/apache/druid/sql/calcite/run/SqlEngine.java

```diff
@@ -116,4 +122,29 @@ QueryMaker buildQueryMakerForInsert(
   default void initContextMap(Map<String, Object> contextMap)
   {
   }
+
+  /**
+   * Returns a {@link SqlStatementFactory} which uses this engine to create statements.
+   */
+  SqlStatementFactory getSqlStatementFactory();
+
+  /**
+   * Returns a list of {@link QueryInfo} containing the currently running queries using this engine.
+   */
+  default List<QueryInfo> getRunningQueries(
+      boolean selfOnly,
+      AuthenticationResult authenticationResult,
+      AuthorizationResult authorizationResult
+  )
+  {
+    throw new UnsupportedOperationException();
```

Comment:
To support the change I asked for in `SqlResource`, this method should return an empty list if the engine does not support "get running queries".